otherlibs/systhreads/st_stubs.c

```diff
@@ -134,24 +134,25 @@ static void caml_thread_scan_roots(scanning_action action,
                                    void *fdata,
                                    caml_domain_state *domain_state)
 {
-  caml_thread_t th;
+  caml_thread_t current, th;
 
-  th = Current_thread;
+  /* If we are called from the backup thread, we cannot access
+     [Current_thread] directly. */
+  current = th = thread_table[Caml_state->id].current_thread;
```

Comment:
`caml_thread_scan_roots` is being passed a `domain_state`. Should it be

```c
current = th = thread_table[domain_state->id].current_thread;
```

Reply:
Nice, thanks.